### DISCUSSION TITLE
OPDS 2 and OPDS-PSE future

### DISCUSSION THREAD

**Discussion by fillito** [anansi-project/opds-pse#11] [General]
I'm sorry if this topic has already been talked about. I couldn't find anything but feel free to point me to it if there is a thread already.

I noticed that OPDS has a [2.0 Draft](https://drafts.opds.io/opds-2.0). 

One of the first things that caught my eye is that it's no longer an XML feed based on Atom, but **JSON** based on [Readium Web Publication Manifest](https://github.com/readium/webpub-manifest). 

- Are there any plans to port OPDS-PSE to JSON so it can be an extension of OPDS 2.0?
- Radium Web Publication Manifest [allows extending](https://github.com/readium/webpub-manifest#8-extensibility) the manifest through profiles. But the extension is expected to be registered and submitted to the official repository. OPDS 2.0 does not appear as an extension of RWPM, though. I don't know whether they will submit it once it's released. 
- Do you guys have plans to support OPDS 2.0 in your servers? @gotson @ajslater @majora2007 @Difegue.

**Comment by Difegue** [comment]
As far as I'm concerned I'll implement the spec if I get any clients to test it against :) Although I didn't know about it before this so nope, no plans! 

It feels like a lot of legacy clients won't implement this so we'll be stuck maintaining PSE 1.x anyways, but it'd be nice to see if we couldn't suggest bundling an implementation of PSE directly in OPDS 2.0 while it's still in draft.

**Reply by Difegue** [reply]
Oh nvm I did see this before, it's the 2019 spec -- it's been in draft for quite a while at this point..

**Comment by gotson** [comment]
I don't think that draft will ever be ready though. If it does, we would need to look into it, but I doubt there would be any server or client implementing any time soon.

I haven't checked the draft in details, so I don't know what it would bring to the table compared to the existing 1.2.

**Reply by HadrienGardeur** [reply]
OPDS 2.0 has been a draft for while but it's very stable and unlikely to change much at this point. It has been pretty widely implemented by ebook/audiobook apps in the library space.

We've started monthly calls once again as part of our efforts to publish updated documents for satellite specs and best practices. Eventually, we'll reach a point where we tackle the OPDS 2.0 draft itself, but I expect this work to be more about spec language and the document that what's defined in there.

**Reply by gotson** [reply]
Hi @HadrienGardeur , good to see you here ! It's definitely an interesting piece of news. Is there a list somewhere with servers/clients implementing the 2.0 draft by any chance ?

I would like also to ask whether there is a validation tool for servers, to verify that the feed is valid according to the spec. This existing for the 1.1 i think.

**Reply by HadrienGardeur** [reply]
OPDS 2.0 is currently supported in Aldiko Next (Android and iOS) and Thorium (Windows, macOS and Linux).

I'm the head of product at De Marque and we've been using OPDS 2.0 for Feedbooks and public libraries for over 3 years now. In Spain for example, most of the ebooks, audiobooks and newspapers borrowed by patrons are delivered through OPDS 2.0.

In the US, LYRASIS and DPLA (Digital Public Library of America) have supported OPDS 2.0 for harvesting metadata for some time now and they're moving their mobile application (Palace, available on Android and iOS) and catalog support for libraries to OPDS 2.0.

Also in the US, the New York Public Library is in the process of migrating to OPDS 2.0 for their entire collection (print + ebooks/audiobooks) in SimplyE (Android and iOS).

There's no online validator for OPDS 2.0 but a JSON Schema has been available for some time now and it's generally used for that purpose: https://drafts.opds.io/schema/feed.schema.json
